CMakeLists.txt

```diff
@@ -90,6 +90,63 @@ list(APPEND rcutils_sources
   include/rcutils/logging_macros.h)
 include_directories("${CMAKE_CURRENT_BINARY_DIR}/include")
 
+###############################################################################
+# Create version file in build/<package>/include/<package> folder which is already in the include list
+set(TMP_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/include/${PROJECT_NAME})
```

Comment:
After being moved into its own package the file should be generated in a directory named after the package providing the CMake function (to avoid potential collisions).

Reply:
@dirk-thomas 

Should I use `rcutils` here of some ament call?

Reply:
Once this logic has been moved into its own package (`ament_cmake_version`) the file should be generated under `${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_version/include/${PROJECT_NAME}`.

Reply:
@dirk-thomas,

I am extremely inept in Ament and its logic to do any changes in it. If you want changes in Ament I think somebody should take over this issue.

Reply:
There should be no need for any knowledge about ament to refactor this logic into a CMake function.

Reply:
fixed: I have refactored the code to a local function `ament_cmake_version`